### ISSUE TITLE
Refactoring results in markdown autocorrect.

### ISSUE THREAD

**Issue by Stedag** [dendronhq/dendron#467]
**Describe the bug**
When refactoring, the current behavior rewrites unrelated text into a strictly defined markdown style. This only occurs in files that are modified by the search and replace process.

**To Reproduce**
1. try refactoring foo to foobar in a vault containing the following example:
```markdown
--------------
- [ ] dashed lines can be useful for separating list components.
--------------
- [ ] dashes are a helpful visual device for documents that you don't intend to preview

if I write a line of text above a barrier another unexpected behavior occurs
---
```
becomes
```markdown

* * *

- [ ] dashed lines can be useful for separating list components.

* * *

- [ ] dashes are a helpful visual device for documents that you don't intend to preview

## if I write a line of text below a barrier another unexpected behavior occurs
```
---
```markdown
...............
- let's try another symbol for a separator and give that a try. This is better, but white-spaces get added between the separator and the newline.
................
- testing another line
...............
```
becomes:
```markdown
...............

- let's try another symbol for a separator and give that a try. This is better, but white-spaces get added between the separator and the newline.
  ................
- testing another line
...............
```
---
to test a few more separator candidates
```markdown

testing
++++

++++
testing

=====
testing

testing
=====

*****
testing

testing
******
```
becomes:
```markdown
testing
\++++

\++++
testing

=====
testing

# testing

* * *

testing

testing

* * *
```

**Expected behavior**
The expected behavior is search and replace.

**Desktop:**
 - Windows 10
 - Vscode 1.52.1
 - Dendron 2.26.1

**Dendron Log file**
The output of `>Dev:Dendron: Open Logs` :
[dendron.log.txt](https://github.com/dendronhq/dendron/files/5877508/dendron.log.txt)

**Comment by kevins8** [comment]
Thanks for the submission. I've noticed this as well. We'll be updating
this behavior in the coming weeks

**Comment by aleksey-rowan** [comment]
This is probably related to https://github.com/dendronhq/dendron/issues/278#issuecomment-712453136 and https://github.com/dendronhq/dendron/issues/572.

**Comment by micharris42** [comment]
Just gave this a plus 1, as the issue still persists.

I used the Refactor Hierarchy command today, and while it only affected a relatively small number of notes (37), within every note affected, my choice of italicization symbols (`*`) were replaced with the Markdown default (`_`).  

An easy fix and hardly the end of the world, but still a bit of a kick in the teeth, especially considering the [Dendron Wiki](https://wiki.dendron.so/notes/eea2b078-1acc-4071-a14e-18299fc28f47.html#refactor-hierarchy) states that "Refactor Hierarchy is a huge time-saver".
